### ISSUE TITLE
[Event Hubs] Investigate Retry surface area

### ISSUE THREAD

**Issue by conniey** [Azure/azure-sdk-for-java#4042]
* The Retry policy has an interesting API surface area. From feedback: https://github.com/Azure/azure-sdk-for-java/pull/3655#discussion_r296132801
  ```java
  class Retry {
     public Duration getNextRetryInterval(Exception lastException, Duration remainingTime);
  }
  ```

**Comment by conniey** [comment]
Move the concern of "lastException" into the consuming class, change int baseWaitTimeSeconds to a Duration.

**Comment by conniey** [comment]
Resolved by #4262
